icu4c/source/common/ustring.cpp

```diff
@@ -1294,7 +1294,15 @@ u_unescapeAt(UNESCAPE_CHAR_AT charAt,
             int32_t ahead = *offset + 1;
             c = charAt(*offset, context);
             if (c == 0x5C /*'\\'*/ && ahead < length) {
-                c = (UChar) u_unescapeAt(charAt, &ahead, length, context);
+                // Calling u_unescapeAt recursively may cause stack overlow if
+                // we have repeated surrogate lead after that. Limit the
+                // length to 5 ('u' and 4 hex) after ahead.
+                int32_t recursive_length = ahead + 5;
```

Comment:
it's not a length, it's a position, right? therefore, i would call it something with "limit" in the name. how about "trailLimit"? (we camelCase variable names)